src/Microsoft.NetCore.Analyzers/Core/Security/DoNotUseWeakKDFInsufficientIterationCount.cs

```diff
@@ -0,0 +1,219 @@
+
+                case PropertySetAbstractValueKind.MaybeFlagged:
+                    return HazardousUsageEvaluationResult.MaybeFlagged;
+
+                default:
+                    return HazardousUsageEvaluationResult.Unflagged;
+            }
+        }
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+
+            // Security analyzer - analyze and report diagnostics on generated code.
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+
+            HazardousUsageEvaluatorCollection hazardousUsageEvaluators = new HazardousUsageEvaluatorCollection(
+                new HazardousUsageEvaluator("GetBytes", HazardousUsageCallback));
+
+            context.RegisterCompilationStartAction(
+                (CompilationStartAnalysisContext compilationStartAnalysisContext) =>
+                {
+                    var wellKnownTypeProvider = WellKnownTypeProvider.GetOrCreate(compilationStartAnalysisContext.Compilation);
+
+                    if (!wellKnownTypeProvider.TryGetTypeByMetadataName(WellKnownTypeNames.SystemSecurityCryptographyRfc2898DeriveBytes, out var rfc2898DeriveBytesTypeSymbol))
+                    {
+                        return;
+                    }
+
+                    var cancellationToken = compilationStartAnalysisContext.CancellationToken;
+                    var sufficientIterationCount = compilationStartAnalysisContext.Options.GetUnsignedIntegralOptionValue(
+                        optionName: EditorConfigOptionNames.SufficientIterationCountForWeakKDFAlgorithm,
+                        rule: DefinitelyUseWeakKDFInsufficientIterationCountRule,
+                        defaultValue: 100000,
+                        cancellationToken: cancellationToken);
+                    var constructorMapper = new ConstructorMapper(
+                        (IMethodSymbol constructorMethod, IReadOnlyList<ValueContentAbstractValue> argumentValueContentAbstractValues,
+                        IReadOnlyList<PointsToAbstractValue> argumentPointsToAbstractValues) =>
+                        {
+                            var kind = PropertySetAbstractValueKind.Flagged;
+
+                            if (constructorMethod.Parameters.Length >= 3)
+                            {
+                                if (constructorMethod.Parameters[2].Name == "iterations" &&
+                                    constructorMethod.Parameters[2].Type.SpecialType == SpecialType.System_Int32)
+                                {
+                                    kind = PropertySetAnalysis.EvaluateLiteralValues(argumentValueContentAbstractValues[2], o => Convert.ToInt32(o) < sufficientIterationCount);
+                                }
+                            }
+
+                            return PropertySetAbstractValue.GetInstance(kind);
+                        });
```

Comment:
Since the iteration threshold is configurable, should check to see if the default of 1000 doesn't meet the threshold.  (Hopefully no one configures it that way... :-) )